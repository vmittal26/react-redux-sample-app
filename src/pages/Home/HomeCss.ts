import styled from 'styled-components';
import { Card } from 'react-bootstrap';

export const HomeContainer = styled(Card)`
  width:80%;
  font-size:0.9rem;
  background-color:#F5F5F6;
  max-width:68rem;
  margin: auto;
  margin-top: 2.8rem;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  height: 90vh;
  padding: 0.5rem;
`;

export const SearchBox = styled.div`
  display: block;
  margin: auto;
  padding: 0.5rem;
  .search {
    display: block;
    margin: 0 auto;
  }
`;

export const PlanetBox = styled.div`
  display: flex;
  flex:1;
  justify-content:flex-start;
  flex-wrap: wrap;
  overflow:auto;
  .spinner-border {
    margin: auto;
  }
`;
