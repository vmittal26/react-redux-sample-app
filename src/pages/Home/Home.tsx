import React, { ReactElement } from 'react';
import { Form, Spinner } from 'react-bootstrap';
import { Message, PlanetItem } from '../../components';
import { NO_PLANETS_FOUND } from '../../shared';
import useSearchPlanets from '../../shared/hooks/useSearchPlanets';
import { HomeContainer, PlanetBox, SearchBox } from './HomeCss';

/**
 * Main page to show planets based on search
 */
export const Home = (): ReactElement => {
  const [searchQuery, setSearchQuey] = React.useState<string>('');
  const { isLoading, planets } = useSearchPlanets(searchQuery);

  const onSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { target: { value } } = event;
    setSearchQuey(value);
  };

  let planetBox = (
    <PlanetBox>
      {searchQuery.trim() !== '' && planets.length > 0 ? (
        planets.map((planet) => <PlanetItem key={planet.name} {...planet} />)
      ) : (
        <Message message={NO_PLANETS_FOUND} />
      )}
    </PlanetBox>
  );

  if (isLoading) {
    planetBox = (
      <PlanetBox>
        <Spinner animation="border" variant="primary" />
      </PlanetBox>
    );
  }
  return (
    <HomeContainer>
      <SearchBox>
        <Form.Control
          className="search"
          type="text"
          placeholder="Search Planets"
          onChange={onSearch}
        />
      </SearchBox>
      {planetBox}
    </HomeContainer>
  );
};
