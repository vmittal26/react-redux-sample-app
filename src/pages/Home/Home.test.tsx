/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';

import { Home } from './Home';
import { render } from '@testing-library/react';

jest.mock('../../shared/hooks/useSearchPlanets');

// const mockUseSearchPlanets = jest.requireMock('useSearchPlanets') as jest.Mock<any>;

describe('test cases for Home', () => {
  it.skip('Home component renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment).toMatchSnapshot();
  });
});
