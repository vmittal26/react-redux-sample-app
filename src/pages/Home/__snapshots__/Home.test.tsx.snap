// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for Home Home component renders correctly 1`] = `[Function]`;
